synapse/handlers/room.py

```diff
@@ -1123,7 +1123,7 @@ async def create_event(
                 event_dict,
                 prev_event_ids=prev_event,
                 depth=depth,
-                state_map=state_map,
+                state_map=dict(state_map),
```

Comment:
I don't understand why we need to take a copy here?

Reply:
When Erik suggested it I think he said something about not mutating the state map but upon your question I looked closer and a: I don't_ see where it's mutated in the functions it's passed to, and b: it's immediately mutated right below so now I am also confused and will clarify it with him.

Reply:
```suggestion
                # Take a copy to ensure each event gets a unique copy of
                # state_map since it is modified below.
                state_map=dict(state_map),
```

Reply:
So is this fixing an actual bug then?

Reply:
Yes, I think so.

Reply:
Should it have a test?!

Reply:
I am judging by the exclamation point that the answer to your question is yes. I hadn't considered that, I will add one.

Reply:
I'm not sure if it is or not, but it seems like a thing we could mistakenly reintroduce by refactoring at some point in the future?

Reply:
I guess it comes down to whether you think the comment is enough of a guard to avoid that - I can see both perspectives, but since the test should be pretty easy to implement I can just add it.

Reply:
Actually now that I've tried to test this I am not sure it's possible - `create_event`, the result of which I am trying to test is a function that is internal to `_send_events_for_new_room`.  `_send_events_for_new_room` consumes the results of `create_event` (an `UnpersistedEventContext` and a new event), converts the `UnpersistedEventContext` into an `EventContext` which then gets sent to the database with the new event, but `EventContexts` aren't stored, so I have no way to access the rather ephemeral `state_map` in question. Note that this bug doesn't affect the actual state or state group for the event. Given this I honestly think that just adding the comment should be fine, but I leave this up to your expertise.

Reply:
If it doesn't get stored in the database then I agree it doesn't seem worth testing.